optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/OptaPlannerProcessor.java

```diff
@@ -103,6 +107,21 @@ FeatureBuildItem feature() {
         return new FeatureBuildItem("optaplanner");
     }
 
+    @BuildStep
+    void registerScoreDirectorFactorySpi(BuildProducer<ServiceProviderBuildItem> services) {
+        String serviceName = ScoreDirectorFactoryProvider.class.getName();
+        String service = "META-INF/services/" + serviceName;
```

Comment:
Doesn't Quarkus have an out-of-the-box way of doing this? Seems low level that we have to mimic this ServiceLoader stuff ourselves. Something to ask on quarkus-dev zulip?

Reply:
This comes straight from Quarkus documentation:
https://quarkus.io/guides/writing-extensions#service-files